netsim/cli/api.py

```diff
@@ -0,0 +1,384 @@
+  handler.end_headers()
+  handler.wfile.write(data)
+
+
+class NetlabHandler(BaseHTTPRequestHandler):
+  def log_message(self, format: str, *args: Any) -> None:
+    return
+
+  def do_GET(self) -> None:
+    if not require_auth(self):
+      return
+    parsed = urllib.parse.urlparse(self.path)
+    parts = [p for p in parsed.path.strip("/").split("/") if p]
+    if not parts:
+      send_json(self, HTTPStatus.NOT_FOUND, {"error": "not found"})
+      return
+
+    if parts == ["healthz"]:
+      send_json(self, HTTPStatus.OK, {"status": "ok"})
+      return
+
+    if parts == ["templates"]:
+      query = urllib.parse.parse_qs(parsed.query)
+      template_dir = query.get("dir", [""])[0]
+      send_json(self, HTTPStatus.OK, {"templates": list_templates(template_dir)})
+      return
+
+    if parts == ["jobs"]:
+      with JOB_LOCK:
+        jobs = list(JOBS.values())
+      send_json(self, HTTPStatus.OK, {"jobs": jobs})
+      return
+
+    if len(parts) >= 2 and parts[0] == "jobs":
+      job_id = parts[1]
+      with JOB_LOCK:
+        job = JOBS.get(job_id)
+      if job is None:
+        send_json(self, HTTPStatus.NOT_FOUND, {"error": "not found"})
+        return
+      if len(parts) == 2:
+        send_json(self, HTTPStatus.OK, job)
+        return
+      if len(parts) == 3 and parts[2] == "log":
+        try:
+          with open(job["logPath"], "r", encoding="utf-8") as fp:
+            content = fp.read()
+        except FileNotFoundError:
+          content = ""
+        send_json(self, HTTPStatus.OK, {"log": content})
+        return
+
+    if parts == ["status"]:
+      out = io.StringIO()
+      with contextlib.redirect_stdout(out), contextlib.redirect_stderr(out):
+        netlab_status.run(["--all"])
+      send_json(self, HTTPStatus.OK, {"status": out.getvalue()})
+      return
+
+    send_json(self, HTTPStatus.NOT_FOUND, {"error": "not found"})
+
+  def do_POST(self) -> None:
+    if not require_auth(self):
+      return
+    parsed = urllib.parse.urlparse(self.path)
+    parts = [p for p in parsed.path.strip("/").split("/") if p]
+    if parts == ["jobs"]:
+      try:
+        payload = parse_json_body(self)
+        job = start_job(payload)
+      except ValueError as exc:
+        send_json(self, HTTPStatus.BAD_REQUEST, {"error": str(exc)})
+        return
+      send_json(self, HTTPStatus.ACCEPTED, job)
+      return
+
+    if len(parts) == 3 and parts[0] == "jobs" and parts[2] == "cancel":
+      job_id = parts[1]
+      with JOB_LOCK:
+        job = JOBS.get(job_id)
+        if job is None:
+          send_json(self, HTTPStatus.NOT_FOUND, {"error": "not found"})
+          return
+        job["state"] = "canceled"
+        job["finishedAt"] = now_iso()
+      send_json(self, HTTPStatus.OK, job)
+      return
+
+    send_json(self, HTTPStatus.NOT_FOUND, {"error": "not found"})
+
+
+def run(cli_args: List[str]) -> None:
+  parser = argparse.ArgumentParser(description="netlab API server")
```

Comment:
All other CLI commands have parser in a separate function to keep the "run" function shorter (and cleaner)